src/Appwrite/Platform/Modules/Proxy/Http/Rules/Create.php

```diff
@@ -0,0 +1,195 @@
+                $message = 'Domain already assigned to different project.';
+            }
+
+            throw new Exception(Exception::RULE_ALREADY_EXISTS, $message);
+        }
+
+        $resourceInternalId = '';
+
+        switch ($resourceType) {
+            case 'function':
+                if (empty($resourceId)) {
+                    throw new Exception(Exception::FUNCTION_NOT_FOUND);
+                }
+
+                $function = $dbForProject->getDocument('functions', $resourceId);
+
+                if ($function->isEmpty()) {
+                    throw new Exception(Exception::RULE_RESOURCE_NOT_FOUND);
+                }
+
+                $resourceInternalId = $function->getInternalId();
+                break;
+            case 'site':
+                if (empty($resourceId)) {
+                    throw new Exception(Exception::SITE_NOT_FOUND);
+                }
+
+                $site = $dbForProject->getDocument('sites', $resourceId);
+
+                if ($site->isEmpty()) {
+                    throw new Exception(Exception::RULE_RESOURCE_NOT_FOUND);
+                }
+
+                $resourceInternalId = $site->getInternalId();
+                break;
+        }
+
+        try {
+            $domain = new Domain($domain);
+        } catch (\Throwable) {
+            throw new Exception(Exception::GENERAL_ARGUMENT_INVALID, 'Domain may not start with http:// or https://.');
+        }
+
+        // TODO: @christyjacob remove once we migrate the rules in 1.7.x
+        $ruleId = System::getEnv('_APP_RULES_FORMAT') === 'md5' ? md5($domain->get()) : ID::unique();
+
+        $rule = new Document([
+            '$id' => $ruleId,
+            'projectId' => $project->getId(),
+            'projectInternalId' => $project->getInternalId(),
+            'domain' => $domain->get(),
+            'resourceType' => $resourceType,
+            'resourceId' => $resourceId,
+            'resourceInternalId' => $resourceInternalId,
+            'certificateId' => '',
+        ]);
+
+        $status = 'created';
+
+        if (\str_ends_with($domain->get(), $functionsDomain) || \str_ends_with($domain->get(), $sitesDomain)) {
+            $status = 'verified';
+        }
+
+        if ($status === 'created') {
+            $target = new Domain(System::getEnv('_APP_DOMAIN_TARGET', ''));
+            $validator = new CNAME($target->get()); // Verify Domain with DNS records
+
+            if ($validator->isValid($domain->get())) {
+                $status = 'verifying';
+
+                $queueForCertificates
+                    ->setDomain(new Document([
+                        'domain' => $rule->getAttribute('domain')
+                    ]))
+                    ->trigger();
+            }
+        }
+
+        $rule->setAttribute('status', $status);
+        $rule = $dbForPlatform->createDocument('rules', $rule);
+
+        $queueForEvents->setParam('ruleId', $rule->getId());
+
+        $rule->setAttribute('logs', '');
```

Comment:
This should not be nessessary here

Reply:
This was part of existing code and still makes sense - we should keep it